docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,122 @@
+
+To verify the `CSIMigrationvSphere` feature flag is enabled and vSphere CSI driver installed on a particular node you can get the CSINode object. You should see `kubernetes.io/vsphere-volume` for an annotation `storage.alpha.kubernetes.io/migrated-plugins`.
+
+    # kubectl get csinodes -o yaml
+    apiVersion: v1
+    items:
+    - apiVersion: storage.k8s.io/v1
+      kind: CSINode
+      metadata:
+        annotations:
+          storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/vsphere-volume
+
+To verify vSphere in-tree PVCs and PVs are migrated to vSphere CSI driver, verify `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotations is present on PVCs and PVs.
+
+PVC Annotations
+
+        Annotations:   pv.kubernetes.io/bind-completed: yes
+                       pv.kubernetes.io/bound-by-controller: yes
+                       pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com
+                       volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/vsphere-volume
+
+PV Annotations
+
+        Annotations:     kubernetes.io/createdby: vsphere-volume-dynamic-provisioner
+                         pv.kubernetes.io/bound-by-controller: yes
+                         pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com
+                         pv.kubernetes.io/provisioned-by: kubernetes.io/vsphere-volume
+
+## Current limitations
+
+- vSphere CSI Driver does not support Provisioning `eagerzeroedthick` and `zeroedthick` volume. After the migration is enabled, when a new volume is requested using the in-tree provisioner and `diskformat` parameter, `diskformat` parameter is dropped by vSphere CSI Driver.
+- vSphere CSI Driver does not support raw vSAN policy parameters. After the migration is enabled, when a new volume is requested using in-tree provisioner and vSAN raw policy parameters specified in the StorageClass, these raw vSAN policy parameters are dropped by vSphere CSI Driver.
+- After in-tree vSphere volume is migrated to vSphere CSI driver, the volume can not be expanded even though CSI driver supports volume expansion.
+- vSphere CSI Migration requires vSphere 7.0u1. Customers who have in-tree vSphere volumes must upgrade vSphere to 7.0u1. Customers who do not need to migrate in-tree vSphere volumes can use vSphere 67u3 and above.
+- Storage Policy used for creating in-tree vSphere volume should not be deleted or renamed. If storage policy is deleted or renamed before migration, it needs to be re-created or re-named with the original name to successfully migrate volume to vSphere CSI driver.
+- Name of the vSphere inventory objects such as Datacenter, Datastore should not be modified after in-tree vSphere volume is provisioned.
+
+## Note
+
+vSphere CSI driver does not support following migration parameters in the StorageClass.
+These parameters are added by vSphere CSI translation library, and should not be used in the stroage class directly.
```

Comment:
IMHO, I think its better we write an admission controller that prevents this. Can you file an issue?

Reply:
Yes, I agree. Filed an issue - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/317

Reply:
added doc about how admission controller helps prevent user from creating or updating StorageClass with these parameters. Also documentation installation steps.